File: backend/src/main/java/com/flickzy/service/interfaces/AuthService.java
package com.flickzy.service.interfaces;

import com.flickzy.dto.AuthResponseDTO;
import com.flickzy.dto.LoginDTO;
import com.flickzy.dto.RegisterDTO;
import com.flickzy.dto.User.UserResponse;

public interface AuthService {
    AuthResponseDTO register(RegisterDTO request);
    AuthResponseDTO login(LoginDTO request);
    void forgotPassword(String email);
    UserResponse resetPassword(String otp, String email, String newPassword);
}

File: frontend/src/services/AuthService.js
import axios from "axios";

export const login = async (email, password) => {
    const response = await axios.post("http://localhost:8386/api/v1/auth/login", {
        email,
        password,
    });
    return response.data;
}

export const register = async (email, password) => {
    const response = await axios.post("http://localhost:8386/api/v1/auth/register", {
        email,
        password,
    });
    return response.data;
}

export const forgotPassword = async (email) => {
    const response = await axios.post("http://localhost:8386/api/v1/auth/forgot-password", {
        email,
    });
    return response.data;
}

export const resetPassword = async (otp, email, newPassword) => {
    const response = await axios.post("http://localhost:8386/api/v1/auth/reset-password", {
        otp,
        email,
        newPassword,
    });
    return response.data;
}